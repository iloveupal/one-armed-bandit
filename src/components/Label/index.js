import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import styled from 'styled-components';

const LabelContainer = styled.div`
    width: 500px;
    height: 30px;
    border-radius: 5px;
    background-color: #27ae60;
    color: white;
    display: flex;
    flex-flow: row no-wrap;
    justify-content: center;
    align-items: center;
    margin: 0 15px;
    font-family: -apple-system, BlinkMacSystemFont, sans-serif;
    font-weight: bold;
`;


export default class Label extends PureComponent {
    static propTypes = {
        title: PropTypes.node.isRequired,
    };

    render () {
        return (
            <LabelContainer>
                <span>{ this.props.title }</span>
            </LabelContainer>
        );
    }
}
