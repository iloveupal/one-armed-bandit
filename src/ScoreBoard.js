import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Label from 'Components/Label';
import Row from 'Components/Row';


export default class ScoreBoard extends PureComponent {
    static propTypes = {
        points: PropTypes.number.isRequired,
    }
    
    render () {
        return (
            <Row>
                <Label
                    title={`You have earned ${this.props.points} points in total!`}
                />
            </Row>
        )
    }
}
