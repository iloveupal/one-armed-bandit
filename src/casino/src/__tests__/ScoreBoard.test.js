import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import ScoreBoard from '../ScoreBoard';
import Label from 'Components/Label';


const getLabel = (points) => `You have earned ${points} points in total!`;

describe('ScoreBoard component', () => {
    it('should match snapshot', () => {
        const component = renderer.create(
            <ScoreBoard
                points={200}
            />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

    it('should reflect the current score on the child label', () => {
        const component = renderer.create(
            <ScoreBoard
                points={200}
            />
        );

        expect(component.root.findByType(Label).props.title).toEqual(getLabel(200));

        component.update(
            <ScoreBoard
                points={201}
            />
        );

        expect(component.root.findByType(Label).props.title).toEqual(getLabel(201));
    });
});