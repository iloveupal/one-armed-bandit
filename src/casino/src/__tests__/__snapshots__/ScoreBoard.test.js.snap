// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreBoard component should match snapshot 1`] = `
.c1 {
  width: 500px;
  height: 30px;
  border-radius: 5px;
  background-color: #27ae60;
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row no-wrap;
  -ms-flex-flow: row no-wrap;
  flex-flow: row no-wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 15px;
  font-family: -apple-system,BlinkMacSystemFont,sans-serif;
  font-weight: bold;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row no-wrap;
  -ms-flex-flow: row no-wrap;
  flex-flow: row no-wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 20px 0;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <span>
      You have earned 200 points in total!
    </span>
  </div>
</div>
`;
